src/traits.rs

```diff
@@ -198,6 +198,18 @@ pub trait Concat<Rhs = Self> {
     fn concat(&self, rhs: &Self) -> Self::Output;
 }
 
+#[cfg(feature = "cross-size")]
+/// Concatenate two numbers into a value of a size equaling the sum of their sizes, using the `rhs`
+/// value as the least significant value.
+pub trait ConcatOther<Rhs> {
+    /// Concatenated output: sum of the widths of `Self` and `Rhs`.
+    type Output;
+
+    /// Concatenate the two values, with `self` as most significant and `rhs`
+    /// as the least significant.
+    fn concat(&self, rhs: &Rhs) -> Self::Output;
+}
```

Comment:
Why are you defining a separate trait for this? Especially one with the exact same shape as the existing `Concat` trait

Reply:
Because I can't use Concat as it specifies `Rhs = Self`.

Reply:
The correct syntax is `Concat<U128>`

Reply:
No, but `Concat` defines `rhs` to be `Self`:
```rust
pub trait Concat<Rhs = Self> {
    /// Concatenated output: twice the width of `Self`.
    type Output;

    /// Concatenate the two values, with `self` as most significant and `rhs`
    /// as the least significant.
    fn concat(&self, rhs: &Self) -> Self::Output;
}
```

Changing the method signature to take `rhs: &Rhs` does allow me to dipose of `ConcatOther`, but I'm not sure whether this is a breaking change or not. I pushed a commit that does this, let me know whether you want this or should I revert.

Reply:
Okay, that's a bug in the definition of `Concat`.

Changing it should be fine.